contracts/scripts/test/smokeTest/mintRedeemTest.js

```diff
@@ -0,0 +1,94 @@
+const { fund, mint } = require('../../../tasks/account');
+const { usdtUnits } = require('../../../test/helpers');
+const addresses = require('../../../utils/addresses');
+const erc20Abi = require('../../../test/abi/erc20.json');
+
+let utils, usdt, ousd, vault, signer
+
+async function fundAccount4(hre) {
+  await fund({
+    num: 1,
+    amount: '2000'
+  }, hre)
+}
+
+const getUsdtBalance = async () => {
+  return await usdt.connect(signer).balanceOf(signer.address)
+}
+
+const getOusdBalance = async () => {
+  return await ousd.connect(signer).balanceOf(signer.address)
+}
+
+async function setup(hre) {
+  utils = hre.ethers.utils
+  ousd = await hre.ethers.getContractAt("OUSD", addresses.mainnet.OUSDProxy);
+  usdt = await hre.ethers.getContractAt(erc20Abi, addresses.mainnet.USDT);
+  const vaultProxy = await hre.ethers.getContract("VaultProxy");
+  vault = await ethers.getContractAt("IVault", vaultProxy.address);
+  signer = (await hre.ethers.getSigners())[4];
+
+  await fundAccount4(hre)
+}
+
+async function beforeDeploy(hre) {
+  // fund stablecoins to the 4th account in signers
+  await setup(hre)
+  
+  const usdtBeforeMint = await getUsdtBalance()
+  const ousdBeforeMint = await getOusdBalance()
+  const usdtToMint = '1000'
+  await mint({
+    num: 1,
+    amount: usdtToMint
+  }, hre)
+
+  const usdtAfterMint = await getUsdtBalance()
+  const ousdAfterMint = await getOusdBalance()
+
+  const expectedUsdt = usdtBeforeMint.sub(usdtUnits(usdtToMint))
+  if (!usdtAfterMint.eq(expectedUsdt)) {
+    throw new Error(`Incorrect usdt value. Got ${usdtAfterMint.toString()} expected: ${expectedUsdt.toString()}`);
+  }
+
```

Comment:
yup had that listed as todo in the code. Since because of the fluctuating value of the stablecoins there needs to be some tolerance applied to the OUSD check. Adding that now.